src/hooks/useDrafts/useDrafts.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useCallback, useContext, useEffect, useState } from 'react'
+import { addDraft, clearDrafts, getDraft, getDrafts, removeDraft, updateDraft } from './utils'
+
+export type CommonDraftProps = {
+  id: string
+  updatedAt?: string
+}
+
+export type DraftType = 'videos' | 'channels'
+
+export type VideoDraft = {
+  type: 'videos'
+  title?: string
+  description?: string
+  publicness?: 'public' | 'private'
+  publishedBefore?: string
+  marketing?: boolean
+  contentRating?: 'all' | 'mature'
+} & CommonDraftProps
+
+export type ChannelDraft = {
+  type: 'channels'
+  title?: string
+  description?: string
+  publicness?: 'public' | 'private'
+  language?: string
+} & CommonDraftProps
+
+export type Draft = VideoDraft | ChannelDraft
+
+type DraftState = {
+  videos: VideoDraft[]
+  channels: ChannelDraft[]
+}
+
+type ContextType = {
+  draftsState: DraftState
+  getInitialDrafts: () => Promise<void>
+}
+
+const DraftsContext = React.createContext<undefined | ContextType>(undefined)
+DraftsContext.displayName = 'DraftsContext'
+
+export const DraftsProvider: React.FC = ({ children }) => {
+  const [draftsState, setDraftsState] = useState<ContextType['draftsState']>({
+    channels: [],
+    videos: [],
+  })
+
+  const getInitialDrafts = useCallback(async () => {
+    try {
+      const currentDrafts = await getDrafts()
+      const videos = currentDrafts.filter((draft): draft is VideoDraft => draft.type === 'videos')
+      const channels = currentDrafts.filter((draft): draft is ChannelDraft => draft.type === 'channels')
+      setDraftsState({
+        videos,
+        channels,
+      })
+    } catch (error) {
+      return Promise.reject(error)
+    }
```

Comment:
Not catching the error will have the same effect (promise rejection) so I don't think this is needed here